In [1]:
import numpy as np
import pandas as pd
import scipy.stats as stats

## Chi Square Test

Mizzare Corporation has collected data on customer satisfaction levels for two types of smart home devices: Smart Thermostats and Smart Lights. They want to determine if there's a significant association between the type of device purchased and the customer's satisfaction level.

In [ ]:
Satisfaction	Smart Thermostat	Smart Light	  Total
Very Satisfied	      50	            70	       120
Satisfied	          80	            100	       180
Neutral	              60	            90	       150
Unsatisfied        	  30	            50	       80
Very Unsatisfied	  20	            50	       70
Total	              240	            360	       600


In [6]:
## Given Data
n=5
df=4
alpha=0.05
CI=0.95

1.State The Hypothesis

H0(Null Hypothesis)=There is no significant association between the type of device purchased and the customer's satisfaction level.
H1(Alternative Hypothesis)=There is a significant association between the type of device purchased and the customer's satisfaction level.

2.Calculate the Chisquare Test

In [11]:
data=np.array([[50,70],[80,100],[60,90],[30,50],[20,50]])

In [13]:
data

array([[ 50,  70],
       [ 80, 100],
       [ 60,  90],
       [ 30,  50],
       [ 20,  50]])

In [15]:
chisqaure_stats=stats.chi2_contingency(data)

In [17]:
chisqaure_stats

Chi2ContingencyResult(statistic=5.638227513227513, pvalue=0.22784371130697179, dof=4, expected_freq=array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]]))

3.Calculate the Critical Value

In [20]:
chi_critical=stats.chi2.ppf(0.95,4)

In [22]:
chi_critical

9.487729036781154

4.Conculusion

1)if 5.638 < 9.487 is fails to reject the null hypothesis,
2)if 5.638 > 9.487 is reject the null hypothesis.
3)Therefore, Null Hypothesis is accepted,
4)There is no significant association between the type of device purchased and the customer's satisfaction level.

## T Test

The theoretical weekly operating cost model: W = $1,000 + $5X
•	Sample of 25 restaurants with a mean weekly cost of Rs. 3,050
•	Number of units produced in a week (X) follows a normal distribution with a mean (μ) of 600 units and a standard deviation (σ) of 25 units


	ˉxˉ = sample mean weekly cost (Rs. 3,050)
•	μ = theoretical mean weekly cost according to the cost model (W = $1,000 + $5X for X = 600 units)
•	σ = 5*25 units
•	n = sample size (25 restaurants)


In [35]:
import numpy as np
import scipy.stats as stats

In [37]:
# Given Data
x_bar=3050
pop_mean=4000
s=125
n=25
alpha=0.05
CI=0.95

1.State the Hpothesis

H0(Null Hypothesis)=The observed mean weekly operating cost is lower than the expected mean weekly operating cost as predicted by the model
H1(Alternative Hypothesis)=The observed mean weekly operating cost is greater than the expected mean weekly operating cost as predicted by the model.

2.Calculate the T Statastic

In [44]:
t_stats=(x_bar-pop_mean)/(s/np.sqrt(n))           

In [46]:
t_stats

-38.0

3.Calculate the Crtical Value

In [49]:
t_critical=stats.t.ppf(0.975,24)

In [51]:
t_critical

2.0638985616280205

4.Conculusion

1)if -38.0 > 2.063 is reject null hypothesis. 2)if -38.0 < 2.063 is fails to reject null hypothesis. 3)HO-Null hypothesis is rejected. 4)Therefore, H1-The observed mean weekly operating cost is greater than the expected mean weekly operating cost as predicted by the model is accepted


